app/models/miq_request_workflow.rb

```diff
@@ -95,9 +96,13 @@ def create_request(values, _requester = nil, auto_approve = false)
 
     request.log_request_success(@requester, :created)
 
-    request.call_automate_event_queue("request_created")
-    request.approve(@requester, "Auto-Approved") if auto_approve == true
-    request.reload if auto_approve
+    if request.process_immediately_on_create?
+      # TODO: address auto-approve potential issue
+      request.call_automate_event_queue("request_created")
+      request.approve(@requester, "Auto-Approved") if auto_approve == true
```

Comment:
@gmcculloug 
All non-service requests will do the auto_approve here. We don't need it for service_order since the service auto_approvals are handled differently.